### PR TITLE
statistics: Refactor code to improve readability

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: None

Problem Summary:

### What is changed and how it works?

- Added more comments to the auto-analyze process.
- Format the code and make sure it is not too long.
- Extracted the `getAllTidsAndPids` to improve readability.
- Removed the useless argument.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢

File: pkg/statistics/handle/autoanalyze/autoanalyze.go [hunk 1]
```diff
@@ -129,36 +132,70 @@ func getAutoAnalyzeParameters(sctx sessionctx.Context) map[string]string {
...
 	statsHandle statsutil.StatsHandle,
-	is infoschema.InfoSchema) (analyzed bool) {
+	is infoschema.InfoSchema,
+) (analyzed bool) {
 	defer func() {
 		if r := recover(); r != nil {
-			logutil.BgLogger().Error("HandleAutoAnalyze panicked", zap.Any("error", r), zap.Stack("stack"))
+			logutil.BgLogger().Error(
+				"HandleAutoAnalyze panicked",
+				zap.Any("error", r),
```
Comment (hawkingrei) [inline pkg/statistics/handle/autoanalyze/autoanalyze.go:160]: We always call this ```r``` as ```recover```.

File: pkg/statistics/handle/autoanalyze/autoanalyze.go [hunk 2]
```diff
@@ -129,36 +132,70 @@ func getAutoAnalyzeParameters(sctx sessionctx.Context) map[string]string {
...
+	// Get the available time period for auto analyze and check if the current time is in the period.
+	start, end, err := parseAnalyzePeriod(
+		parameters[variable.TiDBAutoAnalyzeStartTime],
+		parameters[variable.TiDBAutoAnalyzeEndTime],
+	)
 	if err != nil {
-		logutil.BgLogger().Error("parse auto analyze period failed", zap.String("category", "stats"), zap.Error(err))
+		logutil.BgLogger().Error(
+			"parse auto analyze period failed",
+			zap.String("category", "stats"),
```
Comment (hawkingrei) [inline pkg/statistics/handle/autoanalyze/autoanalyze.go:178]: You can use ```statsLogger``` here.

**Review by qw4990 — APPROVED** [review]

File: pkg/statistics/handle/autoanalyze/autoanalyze.go [hunk 3]
```diff
@@ -227,37 +256,62 @@ func HandleAutoAnalyze(sctx sessionctx.Context,
...
+		ratio,
+	); needAnalyze {
 		escaped, err := sqlescape.EscapeSQL(sql, params...)
 		if err != nil {
 			return false
 		}
-		logutil.BgLogger().Info("auto analyze triggered", zap.String("category", "stats"), zap.String("sql", escaped), zap.String("reason", reason))
+		logutil.BgLogger().Info(
+			"auto analyze triggered",
+			zap.String("category", "stats"),
```
Comment (hawkingrei) [inline pkg/statistics/handle/autoanalyze/autoanalyze.go:302]: Ditto.

File: pkg/statistics/handle/autoanalyze/autoanalyze.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/autoanalyze.go]: Done. Thanks!

File: pkg/statistics/handle/autoanalyze/autoanalyze.go (hunk 2 above)
Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/autoanalyze.go]: Done. Thanks!

File: pkg/statistics/handle/autoanalyze/autoanalyze.go (hunk 3 above)
Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/autoanalyze.go]: Done. Thanks!

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**